packages/components/src/public/QueryModel/GridPanel.tsx

```diff
@@ -240,10 +245,63 @@ class ButtonBar<T> extends PureComponent<GridBarProps<T>> {
     }
 }
 
+interface GridTitleProps {
+    title?: string,
+    model: QueryModel,
+    actions: Actions,
+    allowSelections: boolean,
+    allowViewCustomization: boolean,
+    onRevertView?: () => void,
+}
+
+export const GridTitle: FC<GridTitleProps> = memo(props => {
+    const {title, model, onRevertView, actions, allowSelections, allowViewCustomization } = props;
+    const { queryInfo, viewName } = model;
+
+    let displayTitle = title;
+    let view;
+    if (viewName) {
+        view = queryInfo.views.get(viewName.toLowerCase());
+        if (!view?.hidden) {
+            const label = view.label ?? viewName;
+            displayTitle = displayTitle ? displayTitle + " - " + label : label;
+        }
+    } else {
+        view = queryInfo?.views.get(ViewInfo.DEFAULT_NAME.toLowerCase());
+    }
+    const isEdited = view?.session;
+    const showSave = allowViewCustomization && isEdited && view?.savable;
+    const showRevert = allowViewCustomization && isEdited && view?.revertable;
+
+    if (!displayTitle && !showSave && !isEdited) {
```

Comment:
do we need the `showSave` check in this if statement? I think the `isEdited` part covers that.